docs/markdown/snippets/rust_env_variables.md

```diff
@@ -0,0 +1,22 @@
+## Support for passing environment variables to Rust targets
+
+Rustc idiomatically uses environment variables in macros, such as with
+`include!`:
+
+```rust
+include!(concat!(env!("OUT_DIR"), "/generated.rs"))
+```
+
+Meson will need to support this to effectively wrap cargo dependencies, and for
+idiomatic Rust usage. This may now be done with the `env` keyword argument,
+which works like the one to `custom_target`:
+
+```meson
+ct = custom_target('foo', command : ['foo', '--outdir', '@OUTDIR@'])
+
+executable(
+  'main'
+  'main.rs',
+  env : {'OUT_DIR' : meson.current_build_dir()}
```

Comment:
As far as I know, only Rust is crazy enough to use env at compile time, this should at least be renamed to `rust_env`.